### ISSUE TITLE
Integrate sin times Gaussian does not evaluate

### ISSUE THREAD

**Issue by oscarbenjamin** [sympy/sympy#21362]
```
In [10]: integrate(sin(x)*exp(-x**2), x)
Out[10]: 
⌠               
⎮    2          
⎮  -x           
⎮ ℯ   ⋅sin(x) dx
⌡    
```

**Comment by asmeurer** [comment]
You can get an answer by manually rewriting the expression (this is the same answer given by Wolfram Alpha):

```py
>>> powsimp(expand(sin(x).rewrite(exp)*exp(-x**2)))
I*exp(-x**2 - I*x)/2 - I*exp(-x**2 + I*x)/2
>>> integrate(powsimp(expand(sin(x).rewrite(exp)*exp(-x**2))), x)
-I*sqrt(pi)*exp(-1/4)*erf(x - I/2)/4 + I*sqrt(pi)*exp(-1/4)*erf(x + I/2)/4
```

But if you don't put it in that exact form, it can't do it. I think it is coming from manualintegrate. It's currently bad at detecting that products of exponents can be combined into a single exponential. 

Meijerg is even worse. It can only do this if you manually complete the square inside the exponential (this issue has been discussed many times before in other issues).

```py
>>> integrate(exp(-(x + I/2)**2)/exp(S(1)/4), meijerg=True)
sqrt(pi)*exp(-1/4)*erf(x + I/2)/2
```

So in short, the integrator(s) need to be able to

- Rewrite trig functions as complex exponentials. This is tricky because you only want to do this if the integral can't be computed otherwise, as you might be able to get an answer that doesn't involve complex numbers (e.g., you should *not* do this for the much simpler `integrate(sin(x)*exp(x), x)`).
- Combine products of exponentials into a single exponential.
- Complete the square inside of the exponential if the argument is quadratic. This may apply to other special integrals as well, like `fresnels`.

**Comment by oscarbenjamin** [comment]
> * Complete the square inside of the exponential if the argument is quadratic. This may apply to other special integrals as well, like `fresnels`.

It would be good in general to have a function for completing the square in an expression.

**Comment by asmeurer** [comment]
It would be a useful helper function. Indeed, there are quite a few functions whose derivatives involve `x**2`, and a constant term can be split out, like exponentials with `exp(a + b) = exp(a)*exp(b)`, or for sin(quadratic) you can use `fresnels` and `fresnelc` along with the trig sum identities, or the derivative involves `x**2 + 1`, like `tan(x)` or inverse hyperbolic trig functions. 

An alternative way to view this is that because `-x**2 - I*x = -(x + I/2)**2 - S(1)/4`, one should make the substitution `u = x + I/2`. Doing this (after simplification) completes the square "automatically", in some sense. 

```py
>>> (-x**2 - I*x).subs(x, u - I/2).expand()
-u**2 - 1/4
```

It probably depends on the internals of the specific algorithm, but it may be simpler to apply the shift substitution directly rather than trying to rewrite the expression in a way that the shift can be detected.

**Comment by 11350613** [comment]
With the current main branch I get:
```python
In [1]: str(powsimp(integrate(sin(x)*exp(-x**2), x)))
Out[1]: 'I*(-sqrt(pi)*exp(-1/4)*erf(x - I/2)/4 + sqrt(pi)*exp(-1/4)*erf(x + I/2)/4)'
```

**Comment by oscarbenjamin** [comment]
This was fixed by fed045073d93254b95c9571cee3cd4d9900d6dd7 from gh-28295 which also includes tests for this so this issue can be closed.
